docs/design/sql/08-expressions.md

```diff
@@ -0,0 +1,192 @@
+# SQL Expressions
```

Comment:
Does it make sense to mention `HazelcastOperandTypes.notAny/notAllNull` methods? What is the main motivation to have them? 
For example, why `notAllNull` is used in the `UNARY_MINUS`? Couldn't we convert `-NULL` to just `NULL` without throwing an error?

Reply:
clarified in d1d2929413835f7a462bc88686630dd57203b27c